# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/datamodel/FileSystemDataManager.h
# Commit: 37b7616c8b42
# Full Hash: 37b7616c8b4288e6f92e99ca1314e1f0169d7ed6
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 2
# Description:
#   Bug 1786465 - Convert FileSystemDataManager to be ref counted; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155551
# ==============================================================================

diff -r 399dce373c59 -r 37b7616c8b42 dom/fs/parent/datamodel/FileSystemDataManager.h
--- a/dom/fs/parent/datamodel/FileSystemDataManager.h	Wed Aug 31 09:34:54 2022 +0000
+++ b/dom/fs/parent/datamodel/FileSystemDataManager.h	Wed Aug 31 09:34:54 2022 +0000
@@ -8,6 +8,7 @@
 #define DOM_FS_PARENT_DATAMODEL_FILESYSTEMDATAMANAGER_H_
 
 #include "mozilla/dom/FileSystemTypes.h"
+#include "nsISupportsUtils.h"
 
 namespace mozilla {
 
@@ -18,9 +19,14 @@
 
 class FileSystemDataManager {
  public:
-  using result_t = Result<FileSystemDataManager*, nsresult>;
+  using result_t = Result<RefPtr<FileSystemDataManager>, nsresult>;
   static FileSystemDataManager::result_t CreateFileSystemDataManager(
       const fs::Origin& aOrigin);
+
+  NS_INLINE_DECL_REFCOUNTING(FileSystemDataManager)
+
+ protected:
+  ~FileSystemDataManager() = default;
 };
 
 }  // namespace dom::fs::data
